### PR TITLE
Domains: Don't check availability if input isn't a domain

### PR DESCRIPTION
**Pull request by aidvu**
- More strict validation for search term
- Add maxLength field to SearchCard

Test live: https://calypso.live/?branch=update/fix-suggestion-queries

### CODE REVIEW

File: client/components/search/index.jsx [hunk 1]
```diff
@@ -266,10 +267,13 @@ const Search = React.createClass( {
...
 		const autocorrect = this.props.disableAutocorrect && {
-			autoComplete: 'off',
-			autoCorrect: 'off',
-			spellCheck: 'false'
-		};
+				autoComplete: 'off',
+				autoCorrect: 'off',
+				spellCheck: 'false'
+			},
+			maxLength = this.props.maxLength && {
```
Comment (klimeryk) [inline client/components/search/index.jsx]: I think this is a bit of an overkill and makes it less readable. A simple `maxLength = this.props.maxLength` here and `maxLength = { maxLength }` on the component should suffice. You were probably basing this on the `autocorrect` variable - but that one has more than one property, so it makes sense there :)

Comment (aidvu) [inline client/components/search/index.jsx]: Not really. I couldn't make it work down there. :) And I want it to be conditional.

Comment (aidvu) [inline client/components/search/index.jsx]: Works. HAHA! Thanks! <3

**Comment by aidvu** [comment]
Updated @klimeryk.

**Comment by klimeryk** [comment]
🚢

**Pull request merged**